types/src/front/lib/dust_api.ts

```diff
@@ -379,8 +380,8 @@ export class DustAPI {
       "Content-Type": "application/json",
       Authorization: `Bearer ${this._credentials.apiKey}`,
     };
-    if (groups.length > 0) {
-      headers[DustGroupIdsHeader] = groups.map((g) => g.sId).join(",");
+    if (this._credentials.groups) {
```

Comment:
It feels weird to have this logic hidden and implicit. I think it's the type of logic that even outside of this module you should be aware of, WDYT?

Reply:
Agree that's one way to see it, but for me this is extremely close to being a credential, so I think it's the right place.